lib/acme/certificate.rb

```diff
@@ -9,6 +9,46 @@ def initialize(certificate, chain, request)
     @x509 = certificate
     @x509_chain = chain
     @request = request
+
+    #create_self_signed_cert if x509.nil?
+  end
+
+  def create_self_signed_cert subjectAltDomains
+    keypair = OpenSSL::PKey::RSA.new(2048) 
```

Comment:
Well, IMHO no. The self-signed cert is anyway temporary and you have to copy the certificate PEM to the correct location. I think, at the same cost, one can copy the private key PEM too.